src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -164,6 +164,35 @@ private static void PopulateAllSystemTimeZones(CachedData cachedData)
             }
         }
 
+        private static unsafe string? GetAlternativeId(string id)
+        {
+            char* buffer = stackalloc char[100];
+            if (!GlobalizationMode.Invariant)
```

Comment:
Should we avoid calling this on invariant mode?

Reply:
I am expecting to call these methods when we expose the new APIs too. I preferred to have the check inside the method, so I don't have to check in everywhere we call it. Please let me know if you think otherwise. Thanks!

Reply:
Ah ok. I thought these were going to be the only call sites. Then I think we can leave as is. If we see long term these end up being the only call sites we can adjust.